File: dotnet/CardsProject/Domain/Content/ArticleComment.cs
using CardsProject.Domain.Players;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace CardsProject.Domain.Content;

public class ArticleComment
{
    public int Id { get; set; }

    public string Body { get; set; } = "";
    public bool IsHidden { get; set; } = false;
    [JsonPropertyName("createdAt")]
    public DateTime? CreatedAt { get; set; } = null;

    public int? ArticleId { get; set; }
    [ForeignKey(nameof(ArticleId))]
    public Article? Article { get; set; }
    public int? AuthorId { get; set; }
    [ForeignKey(nameof(AuthorId))]
    public Player? Author { get; set; }
    public int? ParentCommentId { get; set; }
    [ForeignKey(nameof(ParentCommentId))]
    public ArticleComment? ParentComment { get; set; }

    public ICollection<ArticleComment> Replies { get; set; } = new List<ArticleComment>();

    // Business operations

    public void Hide()
    {
        // TODO: implement hide
    }

    public void Unhide()
    {
        // TODO: implement unhide
    }

    public bool IsReply()
    {
        // TODO: implement is_reply
        return default;
    }
}

File: phoenix/lib/cards_project/content/article_comment.ex
defmodule CardsProject.Content.ArticleComment do
  use Ecto.Schema
  import Ecto.Changeset

  schema "article_comments" do
    field :body, :string
    field :is_hidden, :boolean, default: false
    field :created_at, :naive_datetime
    belongs_to :article, CardsProject.Content.Article
    belongs_to :author, CardsProject.Players.Player
    belongs_to :parent_comment, CardsProject.Content.ArticleComment

    timestamps()
  end

  @doc false
  def changeset(record, attrs) do
    record
    |> cast(attrs, [:body, :is_hidden, :created_at, :article_id, :author_id, :parent_comment_id])
    |> validate_required([:body, :is_hidden, :created_at])
  end

  # ── Business operations ────────────────────────────────────────────

  def hide(_record) do
    # TODO: implement ArticleComment.hide
    :ok
  end

  def unhide(_record) do
    # TODO: implement ArticleComment.unhide
    :ok
  end

  def is_reply(_record) do
    # TODO: implement ArticleComment.is_reply
    {:error, :not_implemented}
  end
end
